### DISCUSSION TITLE
Does TabbyML plan on supporting an in-editor Copilot Chat?

### DISCUSSION THREAD

**Discussion by bendavis78** [TabbyML/tabby#2518] [General]
I use [CopilotChat.nvim](https://github.com/CopilotC-Nvim/CopilotChat.nvim) in addition to Copilot.nvim for completion. I find there are many times where completion or comment prompts won't do. 

For example

- "Please explain this function/code block/etc"
- "What are some different approaches to implement this? What are the pros/cons"
- (after selecting a line) "I'm seeing the following error on this line"
- Ability to quickly select & insert changes from the chat
- Ability to provide context, eg `@filename`
- Ability to fix a diagnostic problem from the editor w/ a single keystroke (eg, typing warning, etc)
- Ability to add customized prompts for certain specialized tasks to quickly launch explain/review/fixes

**Comment by wsxiaoys** [comment]
it's already supported in [vscode](https://marketplace.visualstudio.com/items?itemName=TabbyML.vscode-tabby)

<img width="1219" alt="image" src="https://github.com/TabbyML/tabby/assets/388154/d32e4257-b512-4b30-948d-14a0b5a0dcb8">

You'll need to set `--chat-model` to enable the feature, see https://tabby.tabbyml.com/docs/quick-start/installation/docker/ for an example.

**Reply by bendavis78** [reply]
Sorry I hit enter too early on my post. Thanks for the quick reply though! Are these features supported in nvim? 

- "Please explain this function/code block/etc"
- "What are some different approaches to implement this? What are the pros/cons"
- (after selecting a line) "I'm seeing the following error on this line"
- Ability to quickly select & insert changes from the chat
- Ability to provide context, eg `@filename`
- Ability to fix a diagnostic problem from the editor w/ a single keystroke (eg, typing warning, etc)
- Ability to add customized prompts for certain specialized tasks to quickly launch explain/review/fixes

**Reply by wsxiaoys** [reply]
Instead of reimplementing it in Tabby, the best approach is likely to have CopilotChat.nvim support a backend other than GitHub Copilot itself. (Which shouldn't be too challenging, as it [appears](https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/82923efe22b604cf9c0cad0bb2a74aa9247755ab/lua/CopilotChat/copilot.lua#L396) that only the chat/completions and /embeddings endpoints are used).
